aptos-move/aptos-vm/src/transaction_metadata.rs

```diff
@@ -119,6 +121,15 @@ impl TransactionMetadata {
     pub fn is_multi_agent(&self) -> bool {
         !(self.secondary_signers.is_empty() && self.fee_payer.is_none())
     }
+
+    pub fn into_transaction_context(&self) -> TransactionContext {
```

Comment:
Instead implement `Into` and then call `txn_data.into()`, a more idiomatic way to write this in Rust.

Reply:
Good point! But, it's a bit complicated in this case unfortunately. The Rust compiler requires implementation `From` instead of `Into`, and it introduces a cyclic dependency. Since this is only an "into" function, I will leave it as it is until we strongly feel the need for refactoring.